pkg/s3select/helpers.go

```diff
@@ -70,33 +73,34 @@ func representsInt(s string) bool {
 	return false
 }
 
-// The function below processes the where clause into an acutal boolean given a
-// row
-func matchesMyWhereClause(row []string, columnNames map[string]int, alias string, whereClause interface{}) (bool, error) {
-	// This particular logic deals with the details of casting, e.g if we have to
-	// cast a column of string numbers into int's for comparison.
+// matchesMyWhereClause takes map[string]interfaces{} , process the where clause and returns true if the row suffices
+func matchesMyWhereClause(individualRecord map[string]interface{}, alias string, whereClause interface{}) (bool, error) {
 	var conversionColumn string
 	var operator string
 	var operand interface{}
 	if fmt.Sprintf("%v", whereClause) == "false" {
 		return false, nil
 	}
+	out, err := json.Marshal(individualRecord)
+	if err != nil {
+		return false, ErrExternalEvalException
+	}
 	switch expr := whereClause.(type) {
 	case *sqlparser.IsExpr:
-		return evaluateIsExpr(expr, row, columnNames, alias)
+		return evaluateIsExpr(expr, string(out), alias)
 	case *sqlparser.RangeCond:
 		operator = expr.Operator
 		if operator != "between" && operator != "not between" {
 			return false, ErrUnsupportedSQLOperation
 		}
 		if operator == "not between" {
-			myResult, err := evaluateBetween(expr, alias, row, columnNames)
+			myResult, err := evaluateBetween(expr, alias, string(out))
```

Comment:
Can you rename the variable to something like `result` instead of `myResult`?

Reply:
Done.